### PR TITLE
Add FeatherNets 

### PR DESCRIPTION
**Pull request by trushkin-ilya**

### CODE REVIEW

**Comment by trushkin-ilya** [comment]
@grib0ed0v, review the changes please.

File: test.py [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import os
 import torch
+import numpy as np
+import torch.nn.functional as F
+import matplotlib.pyplot as plt
```
Comment (grib0ed0v) [inline test.py]: You can import matplotlib only if visualize is True.
So, you can move this line under if statement.

File: test.py [hunk 2]
```diff
@@ -23,33 +29,94 @@ def evaluate(dataloader: data.DataLoader, model: nn.Module):
...
+        errors.sort(order='prob')
+        errors = np.flip(errors)
+        print(errors)
+        plot_classes_preds(model, zip(*errors))
+
 
     return apcer, bpcer, acer
 
 
+def matplotlib_imshow(img, one_channel=False):
```
Comment (grib0ed0v) [inline test.py]: This function looks like utility, can we create separate file like `utils.py`?

File: test.py [hunk 3]
```diff
@@ -23,33 +29,94 @@ def evaluate(dataloader: data.DataLoader, model: nn.Module):
...
+        img = img.mean(dim=0)
+    img = img / 2 + 0.5  # unnormalize
+    npimg = img.numpy()
+    if one_channel:
+        plt.imshow(npimg, cmap="Greys")
+    else:
+        plt.imshow(np.transpose(npimg, (1, 2, 0)))
+
+
+def images_to_probs(net: nn.Module, images):
```
Comment (grib0ed0v) [inline test.py]: ditto

**Comment by grib0ed0v** [comment]
@trushkin-ilya `models/feathernets.py` file is messy. Can we split it into multiple files and create `feathernet` module for joint model?

File: test.py (hunk 1 above)
Comment (trushkin-ilya) [reply test.py]: But this change will cause `plt` import to be local, so other functions (whole call tree) will require `plt` as parameter to use. Do we actually need this?

File: models/mobile_lite_net.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+        self.layer1 = self._make_layer(block, self.channels[1], layers[0], stride=2)
+        self.layer2 = self._make_layer(block, self.channels[2], layers[1], stride=2)
+        self.layer3 = self._make_layer(block, self.channels[3], layers[2], stride=2)
+        self.layer4 = self._make_layer(block, self.channels[4], layers[3], stride=2)
+        self.final_DW = nn.Conv2d(self.channels[4], self.channels[4], kernel_size=4,
+                                  groups=self.channels[4], bias=False)
+        self.do = nn.Dropout(0.2)
+        self.linear = nn.Linear(self.channels[4] * 16, num_classes)
+        if self.se:
```
Comment (grib0ed0v) [inline models/mobile_lite_net.py]: can we rewrite this?

File: models/mobile_lite_net.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    def forward(self, x):
+        x = self.conv1(x)
+        x = self.bn1(x)
+        x = self.relu(x)
+
+        x = self.conv2(x)
+        x = self.bn2(x)
+        x = self.relu(x)
+
+        if self.se:
```
Comment (grib0ed0v) [inline models/mobile_lite_net.py]: ditto.

File: models/ensemble.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+from torchvision import models
+
+
+class Ensemble(nn.Module):
+    def __init__(self, device, num_classes=2):
+        super(Ensemble, self).__init__()
+
+        self.num_classes = num_classes
+        self.models = [FeatherNetA(num_classes=self.num_classes).to(device),
+                       FeatherNetB(num_classes=self.num_classes).to(device),
```
Comment (grib0ed0v) [inline models/ensemble.py]: Single style instance creation

File: models/feathernets.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+            elif isinstance(m, nn.BatchNorm2d):
+                m.weight.data.fill_(1)
+                m.bias.data.zero_()
+            elif isinstance(m, nn.Linear):
+                n = m.weight.size(1)
+                m.weight.data.normal_(0, 0.01)
+                m.bias.data.zero_()
+
+
+def FeatherNetA(**kwargs):
```
Comment (grib0ed0v) [inline models/feathernets.py]: Do we really need this?

File: models/fishnet.py [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+
+'''
+FishNet
+Author: Shuyang Sun
+'''
+
+__all__ = ['fish']
+
+
+class Fish(nn.Module):
```
Comment (grib0ed0v) [inline models/fishnet.py:7]: FishNet

File: submit.py [hunk 9]
```diff
@@ -87,7 +89,9 @@
         model.eval()
         for mode in ['dev', 'test']:
             dataset = CasiaSurfDataset(protocol, mode=mode, transform=transforms.Compose([
-                transforms.Resize((320, 240)),
+                transforms.Resize(256),
```
Comment (grib0ed0v) [inline submit.py:92]: Other resolutions for exps

File: models/fishnet.py (hunk 8 above)
Comment (trushkin-ilya) [reply models/fishnet.py:7]: there is already `FishNet` class in `fishnet.py`

**Comment by trushkin-ilya** [comment]
@grib0ed0v, waiting for merge approve.

**Review by grib0ed0v — APPROVED** [review]

**Pull request merged**